File: php/src/Model/TaskEscalation.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * Result of advancing the escalation ladder. `escalated: false, parked: true` is the end of the ladder — the task has left matching, and this flag is the only thing that says so.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class TaskEscalation
{
    public function __construct(
        public readonly string $id,
        public readonly bool $escalated,
        public readonly bool $parked,
        public readonly int $escalationLevel,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            escalated: (bool) ($data['escalated'] ?? false),
            parked: (bool) ($data['parked'] ?? false),
            escalationLevel: (int) ($data['escalationLevel'] ?? 0),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/TaskEscalation.java
package io.fivexer.sdk.model;

/**
 * Result of advancing the escalation ladder. {@code escalated: false, parked: true} is the end of the ladder — the task has left matching, and this flag is the only thing that says so.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class TaskEscalation {
    private String id;
    private Boolean escalated;
    private Boolean parked;
    private Integer escalationLevel;

    public String getId() { return id; }
    public Boolean getEscalated() { return escalated; }
    public Boolean getParked() { return parked; }
    public Integer getEscalationLevel() { return escalationLevel; }
}
